bundle/src/main/java/com/adobe/acs/commons/users/impl/ServiceUser.java

```diff
@@ -91,9 +91,9 @@ public String getPrincipalName() {
         return aces;
     }
 
-    public Ace getAce(JackrabbitAccessControlEntry actual) throws RepositoryException {
+    public Ace getAce(JackrabbitAccessControlEntry actual, String path) throws RepositoryException {
         for (Ace ace : getAces()) {
-            if (ace.isSameAs(actual)) {
+            if (StringUtils.equals(path, ace.getContentPath()) && ace.isSameAs(actual)) {
```

Comment:
@drfits food for thought; should isSameAs(..) include this path check implicitly... have to look refresh my brain on this code, but on the surface seems like it should